### DISCUSSION TITLE
training a controlnet - questions and discussion

### DISCUSSION THREAD

**Discussion by jmattthew** [lllyasviel/ControlNet#126] [Q&A]
What kind of target data is needed to reproduce the canny model (as an example)?  Do the target images need to be in a very wide variety of subject matters and styles?  Or could a script be used to generate simple shapes?  Also, do the prompts in the JSON file need to be very descriptive, or can they be simple (e.g. "house")?

I haven't tried training a model on the sample "fill50k" dataset.  But I'm wondering:  will the resulting model only be able to convert circles edges to filled circles and only handle prompts that are similar to its JSON file (e.g. "circles")?   Or will the resulting model be able to convert any shape of edges into solid filled shapes and handle generic prompts (e.g. "house")?

**Comment by geroldmeisinger** [comment]
look here https://github.com/lllyasviel/ControlNet/issues/93
